File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int i=0;
        int j=1;
        int k=1;
        int n=nums.size();

        while(i<n && j<n){
            while(j<n && nums[i]==nums[j]){
                j++;
            }

            if(j<n){
                i++;
                nums[i]=nums[j];
                k++;
            }
        }
        return k;
    }
};

File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int n=nums.length;
        int i=1;
        int k=1;

        while(i<n){
            if(nums[i]!=nums[i-1]){
                nums[k]=nums[i];
                k++;
            }
            i++;
        }
        return k;
    }
}